### PR TITLE
Support for DASH playlists + improve TTML detection

### PR DESCRIPTION
**Pull request by emarsden**
Hello,

MPEG-DASH playlists (an XML format) are used for OTT video streaming. 

The TTML subtitle detection is failing on certain files for which attributes on the root element are not in the same order as the binary fragment tested for; this patch fixes the problem and includes a test.

### CODE REVIEW

**Comment by mmalecot** [comment]
Hello and thank you for your PR!

I will review it ASAP.

In the meantime can you please fix the format warning?

Thanks!

File: src/formats.rs [hunk 1]
```diff
@@ -535,6 +535,13 @@ formats! {
     extension = "dex"
     kind = Executable
 
+    format = DashPlaylist
+    name = "MPEG-DASH manifest"
+    short_name= "DASH"
+    media_type = "application/dash+xml"
+    extension = "dash"
```
Comment (mmalecot) [inline src/formats.rs:542]: I think that the "mpd" extension would be better in this context, as indicated by IANA in its format specification.

File: src/formats.rs [hunk 2]
```diff
@@ -535,6 +535,13 @@ formats! {
     extension = "dex"
     kind = Executable
 
+    format = DashPlaylist
+    name = "MPEG-DASH manifest"
+    short_name= "DASH"
```
Comment (mmalecot) [inline src/formats.rs:540]: "DASH" or "MPD", not sure of the best option.

File: src/formats.rs [hunk 3]
```diff
@@ -535,6 +535,13 @@ formats! {
     extension = "dex"
     kind = Executable
 
+    format = DashPlaylist
```
Comment (mmalecot) [inline src/formats.rs:538]: The variant name must match the name of the file format, e.g. MpegDashMpd if the name is "MPEG-DASH MPD".

**Review by mmalecot — COMMENTED** [review]
Hi,

I finished my review, feel free to react to my comments :).

Thanks!

**Comment by emarsden** [comment]
OK, I think I've addressed all your comments now, thanks.

**Comment by mmalecot** [comment]
Almost, I think you forgot to commit the mpd sample fixture.
Thanks!

**Comment by emarsden** [comment]
I didn't rename it correctly; sorry about that.

**Pull request merged**